lib/mayaUsd/fileio/shading/shadingModeExporterContext.cpp

```diff
@@ -713,6 +718,86 @@ class _UVMappingManager
     MaterialMappings _uvNamesToMaterial;
 };
+
+/// We might have one or more component tags that have roundtrip data that covers this
+// set of faces. Assign them to those tags and return the remaining unhandled faces.
+VtIntArray _AssignComponentTags(
+    const UsdMayaShadingModeExportContext* ctx,
+    const UsdShadeMaterial&                materialToBind,
+    const MObjectHandle&                   geomHandle,
+    const std::vector<UsdGeomSubset>&      currentGeomSubsets,
+    const VtIntArray&                      faceIndices)
+{
+    if (currentGeomSubsets.empty() || faceIndices.empty() || !geomHandle.isValid()) {
+        return faceIndices;
+    }
+
+    JsValue info;
+    if (UsdMayaMeshReadUtils::getGeomSubsetInfo(geomHandle.object(), info) && info) {
+        auto subsetInfoDict = info.GetJsObject();
+
+        std::set<VtIntArray::ElementType> faceSet, handledSet;
+        faceSet.insert(faceIndices.cbegin(), faceIndices.cend());
+
+        MFnDependencyNode shadingGroupDepFn(ctx->GetShadingEngine());
+        std::string       shadingGroupUUID = shadingGroupDepFn.uuid().asString().asChar();
+
+        for (auto&& geomSubset : currentGeomSubsets) {
+            std::string ssName = geomSubset.GetPrim().GetName();
+            const auto  infoIt = subsetInfoDict.find(ssName);
+            if (infoIt != subsetInfoDict.cend()) {
+                // Check candidate:
+                const auto& geomSubsetInfo = infoIt->second.GetJsObject();
+                const auto  uuidIt = geomSubsetInfo.find(UsdMayaGeomSubsetTokens->MaterialUuidKey);
+                if (uuidIt == geomSubsetInfo.cend()
+                    || uuidIt->second.GetString() != shadingGroupUUID) {
+                    continue;
+                }
+
+                VtIntArray subsetIndices;
+                geomSubset.GetIndicesAttr().Get(&subsetIndices);
+
+                // We bind if the componentTag is a subset of the material indices
+                bool bindMaterial = true;
+                for (auto it = subsetIndices.cbegin(); it != subsetIndices.cend(); ++it) {
+                    if (faceSet.count(*it) == 0) {
+                        // The componentTag is out of sync with the material assignment.
+                        // We currently do not try to reconcile as we have no idea how
+                        // the situation happened.
+                        bindMaterial = false;
+                        break;
+                    }
+                }
+
+                if (!bindMaterial) {
+                    continue;
+                }
+
+                if (!ctx->GetExportArgs().exportCollectionBasedBindings) {
+                    UsdShadeMaterialBindingAPI subsetBindingAPI
+                        = UsdMayaTranslatorUtil::GetAPISchemaForAuthoring<
+                            UsdShadeMaterialBindingAPI>(geomSubset.GetPrim());
+                    subsetBindingAPI.Bind(materialToBind);
+                }
```

Comment:
I expected an else... if the flag is not set, nothing is done but the set is still removed as being handled?

Reply:
Same weird code without an else clause as [found in the main function](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/fileio/shading/shadingModeExporterContext.cpp#L806). What is missing is probably the 
```
            if (boundPrimPaths) {
                boundPrimPaths->insert(faceSubset.GetPath());
            }
```
That is used to manage the material collections. Looking at the `exportCollectionBasedBindings` in the unit tests, I do not find much testing of geom subsets.
I can add that, and write a second roundtrip test with that flag lit to make sure it works as I think it should.